app/shared/src/desktopTest/kotlin/ui/VideoControllerStateTest.kt

```diff
@@ -0,0 +1,107 @@
+        state.toggleFullVisible()
+        assertEquals(ControllerVisibility.Invisible, state.visibility)
+        state.toggleFullVisible()
+        assertEquals(ControllerVisibility.Visible, state.visibility)
+        state.toggleFullVisible(true)
+        assertEquals(ControllerVisibility.Visible, state.visibility)
+        state.toggleFullVisible(false)
+        assertEquals(ControllerVisibility.Invisible, state.visibility)
+    }
+
+    @Test
+    fun `show detached slider when init invisible`() = runTest {
+        val state = VideoControllerState(ControllerVisibility.Invisible)
+        val requester = Any()
+        state.setRequestProgressBar(requester)
+        assertEquals(ControllerVisibility.DetachedSliderOnly, state.visibility)
+        state.cancelRequestProgressBarVisible(requester)
+        assertEquals(ControllerVisibility.Invisible, state.visibility)
+
+    }
+
+    @Test
+    fun `do not show detached slider when init visible `() = runTest {
+        val state = VideoControllerState(ControllerVisibility.Visible)
+        val requester = Any()
+        state.setRequestProgressBar(requester)
+        assertEquals(ControllerVisibility.Visible, state.visibility)
+        state.cancelRequestProgressBarVisible(requester)
+        assertEquals(ControllerVisibility.Visible, state.visibility)
+    }
+
+    @Test
+    fun `visibility when nothing`() {
+        val state = createState(false, false, false)
+        assertEquals(ControllerVisibility.Invisible, state.visibility)
+    }
+
+    @Test
+    fun `visibility when alwaysOn`() {
+        val state = createState(true, false, false)
+        assertEquals(ControllerVisibility.Visible, state.visibility)
+    }
+
+    @Test
+    fun `visibility when progressBarVisible`() {
+        val state = createState(false, true, false)
+        assertEquals(ControllerVisibility.DetachedSliderOnly, state.visibility)
+    }
+
+    @Test
+    fun `visibility when fullVisible`() {
+        val state = createState(false, false, true)
+        assertEquals(ControllerVisibility.Visible, state.visibility)
+    }
+
+    @Test
+    fun `visibility when alwaysOn progressBarVisible`() {
+        val state = createState(true, true, false)
+        assertEquals(ControllerVisibility.Visible, state.visibility)
+    }
+
+    @Test
+    fun `visibility when alwaysOn fullVisible`() {
+        val state = createState(true, false, true)
+        assertEquals(ControllerVisibility.Visible, state.visibility)
+    }
+
+    @Test
+    fun `visibility when progressBarVisible fullVisible`() {
+        val state = createState(false, true, true)
+        assertEquals(ControllerVisibility.Visible, state.visibility)
+    }
+
+    @Test
+    fun `visibility when alwaysOn progressBarVisible fullVisible`() {
+        val state = createState(true, true, true)
+        assertEquals(ControllerVisibility.Visible, state.visibility)
+    }
+
+    private fun createState(
```

Comment:
```suggestion
    private fun createStateRequested(
```

因为这个一定会 request 一下 (`setRequestAlwaysOn`), 所以我会倾向于函数名里面就说明